src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -299,6 +299,17 @@ report_status(xrt_core::device_collection& deviceCollection, boost::property_tre
 
 }
 
+static void 
+isSameShellOrSC(DSAInfo& candidate, DSAInfo& current, bool *same_dsa, bool *same_bmc)
+{
+    if (!current.name.empty()) {
+        *same_dsa = ((candidate.name == current.name) &&
```

Comment:
We should never be reaching into a class to retrieve its variables.   

Please create a method to retrieve the `name `and `bmvVer `and compare the resulting values.   

Since we now have methods, please update all of the places where the code reaches into this class to observe these values.